ironmon_tracker/Theme.lua

```diff
@@ -361,38 +446,160 @@ end
 
 function Theme.openExportWindow()
 	Program.destroyActiveForm()
-	local form = forms.newform(515, 125, "Theme Export", function() client.unpause() end)
+	local form = forms.newform(515, 150, "Theme Export", function() client.unpause() end)
 	Program.activeFormId = form
 	Utils.setFormLocation(form, 100, 50)
 
-	local theme_config = Theme.exportThemeToText()
-	forms.label(form, "Copy the theme configuration string below (Ctrl + A --> Ctrl+C):", 9, 10, 300, 20)
-	forms.textbox(form, theme_config, 480, 20, nil, 10, 30)
+	local themeName = Theme.PresetsOrdered[Theme.Screen.currentPreview]
+	local themeCode = Theme.PresetStrings[themeName]
+
+	forms.label(form, "Theme for: " .. themeName, 9, 10, 300, 20)
+	forms.label(form, "Copy the theme code below (Ctrl + A --> Ctrl+C):", 9, 30, 300, 20)
+	forms.textbox(form, themeCode, 480, 20, nil, 10, 55)
 	forms.button(form, "Close", function()
 		forms.destroy(form)
-	end, 212, 55)
+	end, 212, 80)
 end
 
--- Currently unused
 function Theme.openPresetsWindow()
 	Program.destroyActiveForm()
-	local presetsForm = forms.newform(360, 105, "Theme Presets", function() client.unpause() end)
+	local presetsForm = forms.newform(360, 105, "Lookup a Theme Preset", function() client.unpause() end)
 	Program.activeFormId = presetsForm
 	Utils.setFormLocation(presetsForm, 100, 50)
 
-	forms.label(presetsForm, "Select a predefined theme to use:", 49, 10, 250, 20)
+	forms.label(presetsForm, "Select a Theme preset to preview:", 49, 10, 250, 20)
 	local presetDropdown = forms.dropdown(presetsForm, {["Init"]="Loading Presets"}, 50, 30, 145, 30)
 	forms.setdropdownitems(presetDropdown, Theme.PresetsOrdered, false) -- Required to prevent alphabetizing the list
 	forms.setproperty(presetDropdown, "AutoCompleteSource", "ListItems")
 	forms.setproperty(presetDropdown, "AutoCompleteMode", "Append")
 
-	forms.button(presetsForm, "Load", function()
-		Theme.importThemeFromText(Theme.PresetStrings[forms.gettext(presetDropdown)], true)
+	forms.button(presetsForm, "Preview", function()
+		local themeName = forms.gettext(presetDropdown)
+
+		for index, name in ipairs(Theme.PresetsOrdered) do
+			if name == themeName then
+				Theme.Screen.currentPreview = index
+				break
+			end
+		end
+
+		local themeCode = Theme.PresetStrings[Theme.PresetsOrdered[Theme.Screen.currentPreview or 1]]
+		Theme.Buttons.ApplyOrLoadTheme:updateState()
+		Theme.Buttons.RemoveTheme:resetButtonToDefault()
+		Theme.importThemeFromText(themeCode, false)
+
 		client.unpause()
 		forms.destroy(presetsForm)
 	end, 212, 29)
 end
 
+function Theme.openSaveCurrentThemeWindow()
+	Program.destroyActiveForm()
+	local form = forms.newform(350, 145, "Save Theme As...", function() client.unpause() end)
+	Program.activeFormId = form
+	Utils.setFormLocation(form, 100, 50)
+
+	forms.label(form, "Enter a name for this Theme:", 18, 10, 330, 20)
+	local saveTextBox = forms.textbox(form, "", 290, 30, nil, 20, 30)
+	forms.setproperty(saveTextBox, "MaxLength", 80)
```

Comment:
80 is a fine value, from a quick test seems somewhere around 28-ish seems like the limit that fits on the screen (depending on the characters used) if interested in limiting it by that. Disregard if not too concerned about that